code/game/objects/structures/transit_tubes/transit_tube_pod.dm

```diff
@@ -104,52 +113,53 @@
 
 	moving = TRUE
 
-	var/obj/structure/transit_tube/current_tube = null
-	var/next_dir
-	var/next_loc
-	var/last_delay = 0
-	var/exit_delay
-
 	for(var/obj/structure/transit_tube/tube in loc)
 		if(tube.has_exit(dir))
 			current_tube = tube
 			break
 
-	while(current_tube)
-		next_dir = current_tube.get_exit(dir)
+	move_animation(MOVE_ANIMATION_STAGE_ONE)
 
-		if(!next_dir)
-			break
+///timer loop that handles the pod moving from tube to tube
+/obj/structure/transit_tube_pod/proc/move_animation(stage = MOVE_ANIMATION_STAGE_ONE)
+	switch(stage)
+		if(MOVE_ANIMATION_STAGE_ONE)
+			next_dir = current_tube.get_exit(dir)
 
-		exit_delay = current_tube.exit_delay(src, dir)
-		last_delay += exit_delay
+			if(!next_dir)
+				return
 
-		sleep(exit_delay)
+			exit_delay = current_tube.exit_delay(src, dir)
+			next_loc = get_step(loc, next_dir)
 
-		next_loc = get_step(loc, next_dir)
+			current_tube = null
+			for(var/obj/structure/transit_tube/tube in next_loc)
+				if(tube.has_entrance(next_dir))
+					current_tube = tube
+					break
 
-		current_tube = null
-		for(var/obj/structure/transit_tube/tube in next_loc)
-			if(tube.has_entrance(next_dir))
-				current_tube = tube
-				break
+			if(current_tube == null)
+				setDir(next_dir)
+				Move(get_step(loc, dir), dir, DELAY_TO_GLIDE_SIZE(exit_delay)) // Allow collisions when leaving the tubes.
+				return
 
-		if(current_tube == null)
+			enter_delay = current_tube.enter_delay(src, next_dir)
+			if(enter_delay > 0)
+				addtimer(CALLBACK(src, .proc/move_animation, MOVE_ANIMATION_STAGE_TWO), enter_delay)
+				return
+			else
+				return .(MOVE_ANIMATION_STAGE_TWO)
```

Comment:
1. Don't use `.` like this, it's weirdly obfuscating.
2. I don't think we need to go through the overhead of recursively calling this proc. I say make this do `stage = MOVE_ANIMATION_STAGE_TWO`, remove the switch, and make it two separate if statements.

Reply:
done